### ISSUE TITLE
508-defect-2 [SCREENREADER]: "On behalf of" string on the sign as representative field should programmatically describe the field

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#24502]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader in forms mode, I expect descriptive, non-interactive text between interactive form elements (in this case, a field and a checkbox) to be programmatically linked to the objects they are describing so I don't miss them.

## Details

The string "On behalf of [Veteran name]" will most likely be skipped by most screen readers as it is non-interactive text that is sandwiched between two interactive form elements. Because it describes the field "Enter your name to sign as the Veteran's representative," it should be programmatically linked to it using `aria-describedby`.

## Acceptance Criteria

- [ ] Upon tabbing to the field "Enter your name to sign as the Veteran's representative," the screen reader user also hears "on behalf of [Veteran's name]".

## Steps to Recreate
Using a screen reader, tab to the field "Enter your name to sign as the Veteran's representative" and confirm that "on behalf of..." is not announced.

## Proposed Solution (if known)
Use `aria-describedby` to link the "on behalf of..." string to the form field.

## Screenshots or Trace Logs
<img width="1199" alt="Screen Shot 2021-05-10 at 10 49 33 PM" src="https://user-images.githubusercontent.com/14154792/117756432-266ce100-b1ec-11eb-92f7-c9e4cb7b69ca.png">
<img width="1576" alt="Screen Shot 2021-05-11 at 12 06 13 AM" src="https://user-images.githubusercontent.com/14154792/117756801-d80c1200-b1ec-11eb-9c86-ef4926517f82.png">

**Comment by longmd** [comment]
@joshkimux , this one has also been updated. Can you validate that we are good to go?

**Comment by joshkimux** [comment]
That's awesome @longmd ! I'm a little overwhelmed at the moment bandwidth wise, but we're getting some new a11y folk in a couple of weeks or so to help. Tagging in @RLHecht for awareness ❤️

**Comment by longmd** [comment]
@SarahKay8 , I see the issue. The `va-text-input` component doesn't support an `ariaDescribedBy` property like the deprecated React component did. We will have to go back to the Design system team for collaboration to get this property added to the web component.

**Comment by SarahKay8** [comment]
@longmd Gotcha! Please feel free to cc me if needed.

**Comment by longmd** [comment]
@caw310 , is this something we can look at getting added back into the `va-text-input` web component?

**Comment by caw310** [comment]
@longmd we have had a lot of team changes so the DST engineering team is focusing on learning and working on the upgrade to USWDS v3 right now. I will add this to our backlog and can put in as a stretch item in an upcoming sprint. Is this blocking anything?

**Comment by longmd** [comment]
@caw310 , this is not a blocker. A stretch item would be great, thank you!!

**Comment by jamigibbs** [comment]
I did a deep dive on the history of this because it is interesting that the functionality you're looking for here previously existed in the React version of this component.

The TL;DR is that we had made an assumption that the only text content needed for the input field would come from the label but this scenario shows clearly that that is not always the case.

I've added a new prop to `va-text-input` that will allow you to pass in any additional text should be read by the screenreader. It's linked to `aria-describedby` internally. The web component shadow dom will not allow you to pass an element id that is external to the component like the React version would so this is the alternative.

This can be tested at the Chromatic link here: https://github.com/department-of-veterans-affairs/component-library/pull/615

I was able to verify it with Voiceover: 

![217300918-ea44f62a-5722-46e1-9352-1afe59efaaec](https://user-images.githubusercontent.com/872479/217318125-2a0bc4a1-b96a-405f-998a-160ff1a36083.png)

![217300907-f1ea219c-f533-4bf2-9c19-6b16e05daf64](https://user-images.githubusercontent.com/872479/217318132-afdc4a1a-453d-4184-a472-b3eff5941a4e.png)



@SarahKay8 @longmd @joshkimux Are any of you able to verify that this a) works and b) would resolve the issue described in this ticket?

**Comment by SarahKay8** [comment]
@jamigibbs Hi, Jami! Just returned from PTO today. Will take retest in between meetings today.

**Comment by caw310** [comment]
@SarahKay8 do you have time to test this today or on Monday? Thanks!

**Comment by SarahKay8** [comment]
@jamigibbs Validated the new prop is being read via VO+Safari. @longmd  Can we get this updated at your earliest convenience?
This should resolve the issue.

**Comment by longmd** [comment]
That is great news @SarahKay8 ! Thanks for your help with validating the solution. @jamigibbs , just let me know when you get that updated released and I'll make the appropriate updates. 🎉

**Comment by jamigibbs** [comment]
@longmd Once [this PR is merged into vets-website](https://github.com/department-of-veterans-affairs/vets-website/pull/23347), you should have access to that new prop. I'll keep you posted. 👍

**Comment by jamigibbs** [comment]
@longmd Just wanted to give you a head's up that the [latest version of the component library has been merged into vets-website](https://github.com/department-of-veterans-affairs/vets-website/pull/23347) so you're welcome to make that update to your component whenever it's convenient for you.

**Comment by longmd** [comment]
Thanks @jamigibbs !! I really appreciate your effort on this!

**Comment by longmd** [comment]
@SarahKay8 , this is now ready for final signoff. I have tested and get the on behalf of read on Voiceover now.

**Comment by SarahKay8** [comment]
@longmd wonderful!!! Thanks, Matt I'll check this out in a few

**Comment by SarahKay8** [comment]
![IMG_3917](https://user-images.githubusercontent.com/102324990/219708320-1d4e147e-8170-49e3-864a-2a2c96b9c303.jpg)
Validated! Thanks a bunch @jamigibbs @longmd!
